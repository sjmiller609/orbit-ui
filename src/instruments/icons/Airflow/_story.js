import React from 'react'
import { storiesOf } from '@storybook/react'
import { Airflow } from 'instruments'

storiesOf('Instruments|Icons', module).add('Airflow', () => <Airflow />)
