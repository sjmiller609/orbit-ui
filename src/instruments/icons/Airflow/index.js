'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'

import { Box, Icon } from '../../../instruments'

const Airflow = ({ className, dagSize }) => {
  return (
    <Box className={classnames(s.airflow, className)}>
      <Icon icon="airflow_ring" className={classnames(s.border, 'ring')} />
      <Icon icon="dag" className={classnames(s.dag, dagSize)} />
    </Box>
  )
}

Airflow.propTypes = {
  className: PropTypes.string,
  dagSize: PropTypes.string,
}

export default Airflow
