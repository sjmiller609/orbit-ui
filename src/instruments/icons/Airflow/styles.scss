@import 'src/styles/mixins.scss';

$size: 3.6rem;

.airflow {
  height: $size;
  width: $size;
  border-radius: 100rem;
  background: $white;
}

.border {
  height: 100%;
  width: 100%;
  transition: transform 300ms ease-in-out;
}

.dag {
  position: absolute;
  height: $size * 0.4;
  transform: rotate(-35deg);
}
