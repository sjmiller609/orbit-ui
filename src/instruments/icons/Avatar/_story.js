import React from 'react'
import { storiesOf } from '@storybook/react'
import { Avatar } from 'instruments'

storiesOf('Instruments|Icons', module).add('Avatar', () => <Avatar />)
