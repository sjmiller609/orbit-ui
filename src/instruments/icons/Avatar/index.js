'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Icon, Box } from 'instruments'
import s from './styles.scss'

const Avatar = ({ url, title, className }) => {
  return (
    <Box className={classnames(s.avatar, url && s.pic, className)}>
      {url ? (
        <img src={url} title={title} />
      ) : (
        <Icon icon="astro_helmet" title={title} />
      )}
    </Box>
  )
}

Avatar.propTypes = {
  url: PropTypes.string,
  title: PropTypes.string,
  className: PropTypes.string,
}

export default Avatar
