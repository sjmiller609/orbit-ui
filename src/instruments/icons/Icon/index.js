'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import s from './styles.scss'

// const Icon = ({ icon, className }) => {
//   // const src = require(`./${icon}.svg`)
//   // return <img src={src} className={classnames(s.icon, className)} />
//   import(`./${icon}.svg`).then(src => (
//     <img src={src} className={classnames(s.icon, className)} />
//   ))
//   return null
// }

class Icon extends React.Component {
  state = {
    src: null,
  }

  componentWillMount() {
    import(`./${this.props.icon}.svg`).then(src =>
      this.setState({ src: src.default })
    )
  }

  render() {
    const { src } = this.state
    if (!src) return null
    return (
      <img src={src} className={classnames(s.icon, this.props.className)} />
    )
  }
}

Icon.propTypes = {
  icon: PropTypes.string,
  className: PropTypes.string,
}

export default Icon
