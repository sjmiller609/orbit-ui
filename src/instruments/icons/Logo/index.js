'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Link, LoadImg } from 'instruments'

import s from './styles.scss'

const map = {
  A_darkBg: LoadImg(() => import(`./img/A_darkBg.svg`)),
  A_lightBg: LoadImg(() => import(`./img/A_lightBg.svg`)),
  A_stars_darkBg: LoadImg(() => import(`./img/A_stars_darkBg.svg`)),
  A_stars_lightBg: LoadImg(() => import(`./img/A_stars_lightBg.svg`)),
  Astro_darkBg: LoadImg(() => import(`./img/Astro_darkBg.svg`)),
  Astro_lightBg: LoadImg(() => import(`./img/Astro_lightBg.svg`)),
}

class Logo extends React.Component {
  state = {
    Img: null,
  }

  componentWillMount() {
    const { darkBg, noStars, full } = this.props
    let path = full ? 'Astro_' : 'A_'
    path += noStars || full ? '' : 'stars_'
    path += darkBg ? 'darkBg' : 'lightBg'

    this.setState({
      Img: map[path],
    })
  }

  render() {
    const { Img } = this.state
    if (!Img) return null
    const { to, noStars, full, className } = this.props
    return (
      <Link to={to || '/'}>
        <Img
          className={classnames(
            s.logo,
            full && s.full,
            !noStars && s.stars,
            className
          )}
          title="Astronomer"
        />
      </Link>
    )
  }
}

Logo.propTypes = {
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  darkBg: PropTypes.bool,
  full: PropTypes.bool,
  noStars: PropTypes.bool,
  className: PropTypes.string,
}

export default Logo
