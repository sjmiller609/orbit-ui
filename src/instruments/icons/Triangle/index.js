'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import s from './styles.scss'

const Triangle = ({ className }) => {
  return <span className={classnames(s.triangle, className)} />
}

Triangle.propTypes = {
  className: PropTypes.string,
}

export default Triangle
