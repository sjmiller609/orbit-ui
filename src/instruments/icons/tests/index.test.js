import React from 'react'
import { mountWrap } from 'helpers/tests'
import Icon from '../Icon'
import Logo from '../Logo'
import Avatar from '../Avatar'
import Airflow from '../Airflow'
import Triangle from '../Triangle'

const props = {}

const avatarIconProps = {
  title: 'Test',
  className: 'testClassName',
}

const avatarImgProps = {
  url: '/',
  title: 'Test',
  className: 'testClassName',
}

const logoProps = {
  to: '/',
  className: 'textClassName',
}

describe('Icons', () => {
  let icon
  let logo
  let avatar
  let airflow
  let triangle

  beforeEach(() => {
    icon = mountWrap(<Icon {...props} />)
    logo = mountWrap(<Logo {...logoProps} />)
    avatar = mountWrap(<Avatar {...avatarImgProps} />)
    airflow = mountWrap(<Airflow {...props} dagSize="10" />)
    triangle = mountWrap(<Triangle {...props} />)
  })

  it('should be defined', () => {
    expect(Icon).toBeDefined()
    expect(Logo).toBeDefined()
    expect(Avatar).toBeDefined()
    expect(Airflow).toBeDefined()
    expect(Triangle).toBeDefined()
  })

  it('should render correctly', () => {
    expect(icon).toMatchSnapshot()
    expect(logo).toMatchSnapshot()
    expect(avatar).toMatchSnapshot()
    expect(airflow).toMatchSnapshot()
    expect(triangle).toMatchSnapshot()
  })

  it('logo renders full', () => {
    const diffLogo = mountWrap(<Logo full={true} {...logoProps} />)
    expect(diffLogo).toMatchSnapshot()
  })

  it('logo renders darkBg', () => {
    const diffLogo = mountWrap(<Logo darkBg={true} {...logoProps} />)
    expect(diffLogo).toMatchSnapshot()
  })

  it('logo renders stars', () => {
    const diffLogo = mountWrap(<Logo stars={true} {...logoProps} />)
    expect(diffLogo).toMatchSnapshot()
  })

  it('logo defaults to /', () => {
    const diffLogo = mountWrap(<Logo />)
    expect(diffLogo).toMatchSnapshot()
  })

  it('logo returns null if no image', () => {
    const diffLogo = mountWrap(<Logo />)
    diffLogo.setState({ Img: null })
    expect(diffLogo).toMatchSnapshot()
  })

  it('avatar should check for url and load image', () => {
    const conditionalIcon = avatar.find(Icon).length
    expect(conditionalIcon).toBe(0)
  })

  it('avatar should check for url and load icon', () => {
    const avatarNoIcon = mountWrap(<Avatar {...avatarIconProps} />)
    const conditionalIcon = avatarNoIcon.find(Icon).length
    expect(conditionalIcon).toBe(1)
  })
})
