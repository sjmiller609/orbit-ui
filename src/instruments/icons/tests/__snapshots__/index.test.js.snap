// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons logo defaults to / 1`] = `
<withRouter(Logo)>
  <Route>
    <Logo
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
    >
      <Link
        onClick={[Function]}
      >
        <A
          className="link"
          onClick={[Function]}
        >
          <a
            className="link"
            onClick={[Function]}
          >
            <Imported
              className="logo"
              title="Astronomer"
            >
              <ReactImportedComponent
                className="logo"
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                title="Astronomer"
              >
                <UnconnectedReactImportedComponent
                  async={false}
                  className="logo"
                  exportPicker={[Function]}
                  loadable={
                    Object {
                      "done": true,
                      "importFunction": [Function],
                      "load": [Function],
                      "mark": Array [],
                      "ok": true,
                      "payload": Object {
                        "default": Object {
                          "process": [Function],
                        },
                        "process": [Function],
                      },
                      "promise": Promise {},
                      "reset": [Function],
                    }
                  }
                  render={[Function]}
                  streamId={0}
                  title="Astronomer"
                >
                  <img
                    className="logo"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                    title="Astronomer"
                  />
                </UnconnectedReactImportedComponent>
              </ReactImportedComponent>
            </Imported>
          </a>
        </A>
      </Link>
    </Logo>
  </Route>
</withRouter(Logo)>
`;

exports[`Icons logo renders darkBg 1`] = `
<withRouter(Logo)
  className="textClassName"
  darkBg={true}
  to="/"
>
  <Route>
    <Logo
      className="textClassName"
      darkBg={true}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
      to="/"
    >
      <Link
        onClick={[Function]}
      >
        <A
          className="link"
          onClick={[Function]}
        >
          <a
            className="link"
            onClick={[Function]}
          >
            <Imported
              className="logo textClassName"
              title="Astronomer"
            >
              <ReactImportedComponent
                className="logo textClassName"
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                title="Astronomer"
              >
                <UnconnectedReactImportedComponent
                  async={false}
                  className="logo textClassName"
                  exportPicker={[Function]}
                  loadable={
                    Object {
                      "done": true,
                      "importFunction": [Function],
                      "load": [Function],
                      "mark": Array [],
                      "ok": true,
                      "payload": Object {
                        "default": Object {
                          "process": [Function],
                        },
                        "process": [Function],
                      },
                      "promise": Promise {},
                      "reset": [Function],
                    }
                  }
                  render={[Function]}
                  streamId={0}
                  title="Astronomer"
                >
                  <img
                    className="logo textClassName"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                    title="Astronomer"
                  />
                </UnconnectedReactImportedComponent>
              </ReactImportedComponent>
            </Imported>
          </a>
        </A>
      </Link>
    </Logo>
  </Route>
</withRouter(Logo)>
`;

exports[`Icons logo renders full 1`] = `
<withRouter(Logo)
  className="textClassName"
  full={true}
  to="/"
>
  <Route>
    <Logo
      className="textClassName"
      full={true}
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
      to="/"
    />
  </Route>
</withRouter(Logo)>
`;

exports[`Icons logo renders stars 1`] = `
<withRouter(Logo)
  className="textClassName"
  stars={true}
  to="/"
>
  <Route>
    <Logo
      className="textClassName"
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
      stars={true}
      to="/"
    >
      <Link
        onClick={[Function]}
      >
        <A
          className="link"
          onClick={[Function]}
        >
          <a
            className="link"
            onClick={[Function]}
          >
            <Imported
              className="logo textClassName"
              title="Astronomer"
            >
              <ReactImportedComponent
                className="logo textClassName"
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                title="Astronomer"
              >
                <UnconnectedReactImportedComponent
                  async={false}
                  className="logo textClassName"
                  exportPicker={[Function]}
                  loadable={
                    Object {
                      "done": true,
                      "importFunction": [Function],
                      "load": [Function],
                      "mark": Array [],
                      "ok": true,
                      "payload": Object {
                        "default": Object {
                          "process": [Function],
                        },
                        "process": [Function],
                      },
                      "promise": Promise {},
                      "reset": [Function],
                    }
                  }
                  render={[Function]}
                  streamId={0}
                  title="Astronomer"
                >
                  <img
                    className="logo textClassName"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                    title="Astronomer"
                  />
                </UnconnectedReactImportedComponent>
              </ReactImportedComponent>
            </Imported>
          </a>
        </A>
      </Link>
    </Logo>
  </Route>
</withRouter(Logo)>
`;

exports[`Icons should render correctly 1`] = `<Icon />`;

exports[`Icons should render correctly 2`] = `
<withRouter(Logo)
  className="textClassName"
  to="/"
>
  <Route>
    <Logo
      className="textClassName"
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={Object {}}
      match={Object {}}
      to="/"
    >
      <Link
        onClick={[Function]}
      >
        <A
          className="link"
          onClick={[Function]}
        >
          <a
            className="link"
            onClick={[Function]}
          >
            <Imported
              className="logo textClassName"
              title="Astronomer"
            >
              <ReactImportedComponent
                className="logo textClassName"
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                title="Astronomer"
              >
                <UnconnectedReactImportedComponent
                  async={false}
                  className="logo textClassName"
                  exportPicker={[Function]}
                  loadable={
                    Object {
                      "done": true,
                      "importFunction": [Function],
                      "load": [Function],
                      "mark": Array [],
                      "ok": true,
                      "payload": Object {
                        "default": Object {
                          "process": [Function],
                        },
                        "process": [Function],
                      },
                      "promise": Promise {},
                      "reset": [Function],
                    }
                  }
                  render={[Function]}
                  streamId={0}
                  title="Astronomer"
                >
                  <img
                    className="logo textClassName"
                    src={
                      Object {
                        "process": [Function],
                      }
                    }
                    title="Astronomer"
                  />
                </UnconnectedReactImportedComponent>
              </ReactImportedComponent>
            </Imported>
          </a>
        </A>
      </Link>
    </Logo>
  </Route>
</withRouter(Logo)>
`;

exports[`Icons should render correctly 3`] = `
<Avatar
  className="testClassName"
  title="Test"
  url="/"
>
  <Box
    className="avatar pic testClassName"
  >
    <Flex
      className="avatar pic testClassName"
      flow="box"
    >
      <div
        className="flex box undefined avatar pic testClassName"
      >
        <img
          src="/"
          title="Test"
        />
      </div>
    </Flex>
  </Box>
</Avatar>
`;

exports[`Icons should render correctly 4`] = `
<Airflow
  dagSize="10"
>
  <Box
    className="airflow"
  >
    <Flex
      className="airflow"
      flow="box"
    >
      <div
        className="flex box undefined airflow"
      >
        <Icon
          className="border ring"
          icon="airflow_ring"
        >
          <Imported
            className="border ring"
          >
            <ReactImportedComponent
              className="border ring"
              loadable={
                Object {
                  "done": true,
                  "importFunction": [Function],
                  "load": [Function],
                  "mark": Array [],
                  "ok": true,
                  "payload": Object {
                    "default": Object {
                      "process": [Function],
                    },
                    "process": [Function],
                  },
                  "promise": Promise {},
                  "reset": [Function],
                }
              }
              render={[Function]}
            >
              <UnconnectedReactImportedComponent
                async={false}
                className="border ring"
                exportPicker={[Function]}
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                streamId={0}
              >
                <img
                  className="border ring"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              </UnconnectedReactImportedComponent>
            </ReactImportedComponent>
          </Imported>
        </Icon>
        <Icon
          className="dag 10"
          icon="dag"
        >
          <Imported
            className="dag 10"
          >
            <ReactImportedComponent
              className="dag 10"
              loadable={
                Object {
                  "done": true,
                  "importFunction": [Function],
                  "load": [Function],
                  "mark": Array [],
                  "ok": true,
                  "payload": Object {
                    "default": Object {
                      "process": [Function],
                    },
                    "process": [Function],
                  },
                  "promise": Promise {},
                  "reset": [Function],
                }
              }
              render={[Function]}
            >
              <UnconnectedReactImportedComponent
                async={false}
                className="dag 10"
                exportPicker={[Function]}
                loadable={
                  Object {
                    "done": true,
                    "importFunction": [Function],
                    "load": [Function],
                    "mark": Array [],
                    "ok": true,
                    "payload": Object {
                      "default": Object {
                        "process": [Function],
                      },
                      "process": [Function],
                    },
                    "promise": Promise {},
                    "reset": [Function],
                  }
                }
                render={[Function]}
                streamId={0}
              >
                <img
                  className="dag 10"
                  src={
                    Object {
                      "process": [Function],
                    }
                  }
                />
              </UnconnectedReactImportedComponent>
            </ReactImportedComponent>
          </Imported>
        </Icon>
      </div>
    </Flex>
  </Box>
</Airflow>
`;

exports[`Icons should render correctly 5`] = `
<Triangle>
  <span
    className="triangle"
  />
</Triangle>
`;
